kythe/go/extractors/gcp/config/converter.go

```diff
@@ -120,7 +110,28 @@ func readConfigFile(input string) (*rpb.Config, error) {
 	return conf, nil
 }
 
-func initializeArtifacts(build *cloudbuild.Build) {
-	build.Artifacts = &cloudbuild.Artifacts{}
-	build.Artifacts.Objects = &cloudbuild.ArtifactObjects{}
+// buildSystemStepper encapsulates the logic for adding extra build steps for a
+// specific type of build system (maven, gradle, etc).
+// TODO(danielmoy): we will almost certainly need to support more fine-grained
+// steps and/or artifacts.  For example now artifacts are prepended, we might
+// need ones that are appended.  We might need build steps that occur before or
+// directly after cloning, but before other common steps.
+type buildSystemStepper interface {
```

Comment:
I don't really like "stepper". Is there something else you could use? buildSystemConfig? buildSystemSteps? Something else?

Reply:
buildStepElaborator

Reply:
sorry, buildSystemElaborator

Reply:
ah sorry I didn't see that.  I had buildStepGenerator

buildSystemElaborator might make more sense, especially if we end up needing to throw more things than steps and one artifact - timeouts and secrets I think could be featured in here in the future